executor/infoschema_reader.go

```diff
@@ -2615,54 +2613,73 @@ type tiflashInstanceInfo struct {
 }
 
 func (e *TiFlashSystemTableRetriever) initialize(sctx sessionctx.Context, tiflashInstances set.StringSet) error {
-	store := sctx.GetStore()
-	if etcd, ok := store.(kv.EtcdBackend); ok {
-		var addrs []string
-		var err error
-		if addrs, err = etcd.EtcdAddrs(); err != nil {
+	storeInfo, err := infoschema.GetStoreServerInfo(sctx)
+	if err != nil {
+		return err
+	}
+
+	for _, info := range storeInfo {
+		if info.ServerType != kv.TiFlash.Name() {
+			continue
+		}
+		info.ResolveLoopBackAddr()
+		if len(tiflashInstances) > 0 && !tiflashInstances.Exist(info.Address) {
+			continue
+		}
+		hostAndStatusPort := strings.Split(info.StatusAddr, ":")
+		if len(hostAndStatusPort) != 2 {
+			return errors.Errorf("node status addr: %s format illegal", info.StatusAddr)
+		}
+		// fetch tiflash config
+		configURL := fmt.Sprintf("%s://%s/config", util.InternalHTTPSchema(), info.StatusAddr)
+		resp, err := util.InternalHTTPClient().Get(configURL)
+		if err != nil {
```

Comment:
Looks like if all TiFlash instances return error for this call, the error will be ignored, is it an excepted behavior?

Reply:
Yes, it is expected to just silently ignore all tiflash instance that return error. Related issue: https://github.com/pingcap/tidb/issues/18701